### PR TITLE
bpo-40492: Fix `--outfile` when the program being profiled changes the working directory

### PR DESCRIPTION
**Pull request by asottile**
https://bugs.python.org/issue40492

### CODE REVIEW

File: Lib/cProfile.py [hunk 1]
```diff
@@ -152,6 +152,11 @@ def main():
     (options, args) = parser.parse_args()
     sys.argv[:] = args
 
+    # the script that we're profiling may chdir, so capture the absolute path
+    # to the output file at startup
```
Comment (taleinat) [inline Lib/cProfile.py]: As per [pep8](https://www.python.org/dev/peps/pep-0008/#comments), please start comments with a capital letter and end with a period.
```suggestion
    # The script that we're profiling may chdir, so capture the absolute path
    # to the output file at startup.
```

File: Lib/profile.py [hunk 2]
```diff
@@ -571,6 +571,11 @@ def main():
     (options, args) = parser.parse_args()
     sys.argv[:] = args
 
+    # the script that we're profiling may chdir, so capture the absolute path
+    # to the output file at startup
```
Comment (taleinat) [inline Lib/profile.py]:
```suggestion
    # The script that we're profiling may chdir, so capture the absolute path
    # to the output file at startup.
```

File: Misc/NEWS.d/next/Library/2020-05-04-12-16-00.bpo-40492.ONk9Na.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Fix ``--outfile`` for :mod:`cProfile` / :mod:`profile` when the program being
+profiled changes the working directory - by Anthony Sottile.
```
Comment (taleinat) [inline Misc/NEWS.d/next/Library/2020-05-04-12-16-00.bpo-40492.ONk9Na.rst]:
```suggestion
Fix ``--outfile`` for :mod:`cProfile` / :mod:`profile` not writing the output
file in the original directory when the program being profiled changes the
working directory.  PR by Anthony Sottile.
```

**Review by taleinat — CHANGES_REQUESTED** [review]
IMO this is an excellent fix! Just a few minor style and wording requests.

**Comment by asottile** [comment]
the suggestions seem nitpicky at best, but I've added them -- it seems the devguide has changed at least twice since I've started committing to cpython (I went through my PRs and there's ~3 different styles that were suggested for attribution in them (!))

**Comment by asottile** [comment]
I didn't expect the Spanish Inquisition

**Comment by taleinat** [comment]
> the suggestions seem nitpicky at best, but I've added them

Thanks! It is appreciated.

The (duplicate) one about the comments is what it is - we have a style guide and we try to stick to it. Yes, it can feel like nitpicking sometimes. I do try to use GitHub's "suggestion" feature to allow applying such suggestions with a click, to reduce the adoing so as easy as possible.nnoyance as much as possible.

> I went through my PRs and there's ~3 different styles that were suggested for attribution in them (!)

That attribution at the end of the NEWS entry wouldn't have needed to be changed by itself, it was fine. The meat of that change request was to clarify what was fixed. I only changed the attribution in my suggestion out of habit.

**Comment by asottile** [comment]
> 
> 
> > the suggestions seem nitpicky at best, but I've added them
> 
> Thanks! It is appreciated.
> 
> The (duplicate) one about the comments is what it is - we have a style guide and we try to stick to it. Yes, it can feel like nitpicking sometimes. I do try to use GitHub's "suggestion" feature to allow applying such suggestions with a click, to make doing so as easy as possible.
> 
> > I went through my PRs and there's ~3 different styles that were suggested for attribution in them (!)
> 
> That attribution at the end of the NEWS entry wouldn't have needed to be changed by itself, it was fine. The meat of that change request was to clarify what was fixed. I only changed the attribution in my suggestion out of habit.

just seems like a waste of time after this has been sitting here for 6 months -- I don't really agree with your change to my changelog entry either but given it's already been sitting here without mention for 6 months I simply don't have time to argue about it

cpython "adheres to PEP8" in that sometimes it's mentioned in review but the vast majority does not adhere (and certainly not to all of it) -- for instance Lib/profile.py has 8 (!) such comments which don't match your suggestion!  I don't run this project but I'd usually just ignore things (or invest in tooling to automatically report / fix them) unless they are actually broken architecturally and even then I'd probably just fix them myself by pushing to the PR 🤷 

either way, we're at a place where I think (?) you're happy with the change

**Review by taleinat — APPROVED** [review]

**Comment by taleinat** [comment]
> just seems like a waste of time after this has been sitting here for 6 months

I know what that feels like and I'm sorry for that. Such delays are a big problem, and it's something that I'm personally working to address, as well as several others in the core dev team and in the PSF.

> I'd usually just ignore things (or invest in tooling to automatically report / fix them)

With this I agree, I don't like spending my own time checking code style either, it is not a good investment of time. I intend to bring this up in the upcoming core dev sprint in terms of improving the experience of contributing to the project.

> I'd probably just fix them myself by pushing to the PR

I avoid making such changes directly since for many people that can feel like too direct intervention in their own work. I'll try to remember that you prefer that I just make such changes directly on your PRs.

Again, thanks for the PR! :pray:

**Pull request merged**